src/vendor/detectElementResize.js

```diff
@@ -10,46 +10,84 @@
  * 4) Add nonce for style element.
  **/
 
-export default function createDetectElementResize(nonce) {
-  // Check `document` and `window` in case of server-side rendering
-  var _window;
-  if (typeof window !== 'undefined') {
-    _window = window;
-  } else if (typeof self !== 'undefined') {
-    _window = self;
-  } else {
-    _window = global;
-  }
+// Check `document` and `window` in case of server-side rendering
+let windowObject;
+if (typeof window !== 'undefined') {
+  windowObject = window;
+
+  // eslint-disable-next-line no-restricted-globals
+} else if (typeof self !== 'undefined') {
+  // eslint-disable-next-line no-restricted-globals
+  windowObject = self;
+} else {
+  windowObject = global;
+}
+
```

Comment:
not sure if using Flow/TS, but is there any scenario where the window object would still be null here?

Reply:
No. The `else` here falls back to global when `window` and `self` are undefined. One of the three should always be present.